src/corelibs/U2Gui/src/Notification.cpp

```diff
@@ -453,4 +356,68 @@ bool NotificationStack::eventFilter(QObject* target, QEvent* event) {
     return false;
 }
 
+void NotificationStack::activate(Notification* notification) {
+    SAFE_POINT(notification, "Notification is null", );
+    QString text = notification->text;
+    QAction* action = notification->action;
+    if (floatingNotifications.contains(notification)) {
+        remove(notification);
+    }
+    if (action) {
+        action->trigger();
+        return;
+    }
+
+    QObjectScopedPointer<QDialog> dlg = new QDialog(AppContext::getMainWindow()->getQMainWindow());
+    dlg->setObjectName("NotificationDialog");
+    QVBoxLayout vLayout;
+    QHBoxLayout hLayout;
+    QPushButton ok;
+    QCheckBox isDelete;
+
+    ok.setText(tr("OK"));
+    isDelete.setText(tr("Remove notification after closing"));
+    isDelete.setChecked(true);
+    connect(&ok, SIGNAL(clicked()), dlg.data(), SLOT(accept()));
```

Comment:
Old `SIGNAL` and `SLOT`

Reply:
This method was moved from Notification to NotificationStack as is. There is no my changes in this code.